doc/fish-completion.fish

```diff
@@ -0,0 +1,164 @@
+# fish completion for restic                               -*- shell-script -*-
```

Comment:
That completion file should be generated during the release process. Please extend `helpers/prepare-release/main.go` accordingly and remove this file.